gc/base/Math.hpp

```diff
@@ -137,6 +139,61 @@ class MM_Math
 		return number;
 	}
 
+
+	/**
+	 * Helper function to convert size to index using integers.
+	 * In general conversion looks like index = logBase(size).
+	 * Using log2(), this conversion can be written as index = log2(size)*(1/log2(base)).
+	 * This function is hard coded to use 4 as 1/log2(base). It is corresponds with base = 1.18920712.
+	 * Such selected value allows to use approximated log2() * 4 value as an index.
+	 * Calculation of log2() is approximated:
+	 * - integer part of log2() is calculated as MSB position;
+	 * - next two bits are used for fractional part calculation (0,0.25,0.5,0.75).
+	 * MSB position is calculated as topBit position minus number of leading zeroes.
+	 * Number of leading zeroes is returned by MM_Bits::trailingZeroes().
+	 * Please note, there is limitation for input size >= 8. Caller sites should control this.
+	 * Please note this function supports 64 and 32 bit platforms.
+	 *
+	 * @param size needs to be converted to the index.
+	 * @return index correspondent to the size.
+	 */
+	MMINLINE static uintptr_t sizeToIndex(uintptr_t size)
+	{
+		uintptr_t msb = (J9BITS_BITS_IN_SLOT - 1) - MM_Bits::trailingZeroes(size);
```

Comment:
out of scope for this change, but should eventually rename this macro to OMRBITS_BITS_IN_SLOT